### PR TITLE
Configure pulseaudio d-bus access to bluez

### PR DESCRIPTION
**Pull request by kris-lab**
Following policy allows the custom `pulseaudio` service to talk to `bluez`.

`/etc/dbus-1/system.d/pulseaudio-system.conf`
```
  <policy user="user-name">
    <allow send_destination="org.bluez"/>
    <allow send_interface="org.bluez.Manager"/>
    <allow send_interface="org.bluez.Agent"/>
    <allow send_interface="org.bluez.HandsfreeAgent"/>
    <allow send_interface="org.bluez.MediaEndpoint"/>
    <allow send_interface="org.bluez.MediaPlayer"/>
    <allow send_interface="org.bluez.Watcher"/>
    <allow send_interface="org.bluez.ThermometerWatcher"/>
  </policy>
```

### CODE REVIEW

**Comment by kris-lab** [comment]
@ppp0 have a look on skeleton and structure. I am not sure where to put this configuration so I decided to abstract `dbus` into module and use it in the `pulseaudio:service`.

cc @njam

File: modules/pulseaudio/templates/dbus [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<policy user="<%= @user %>">
+  <allow send_destination="org.bluez"/>
+  <allow send_interface="org.bluez.Manager"/>
+  <allow send_interface="org.bluez.Agent"/>
```
Comment (ppp0) [inline modules/pulseaudio/templates/dbus]: We only need interface `org.bluez.Manager`

File: modules/dbus/manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class dbus {
+
+  require 'apt'
+
+  package { 'dbus':
+    provider => apt,
+  }
+
+  service { 'dbus':
```
Comment (ppp0) [inline modules/dbus/manifests/init.pp]: Better do not touch the service..

File: modules/dbus/spec/entry_systemd/spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+describe 'dbus::service' do
+
+  describe service('dbus') do
```
Comment (ppp0) [inline modules/dbus/spec/entry_systemd/spec.rb]: Don't check service, check policy file

File: modules/dbus/manifests/entry/systemd.pp [hunk 4]
```diff
@@ -0,0 +1,14 @@
+define dbus::entry::systemd (
```
Comment (ppp0) [inline modules/dbus/manifests/entry/systemd.pp]: dbus::config::system?

**Comment by ppp0** [comment]
- Ok, as discussed, let's have a minimal wrapper class `dbus` (Dbus being a dependency of `systemd`among a lot others, it will be there...)
- Please see comments to the diffs

**Comment by kris-lab** [comment]
@ppp0 please re-review

**Comment by ppp0** [comment]
* Do we require `class pulseaudio::module::x11` anywhere? Will this be happening later?
Otherwise LGTM!

**Comment by kris-lab** [comment]
Well I am gonna require this https://github.com/cargomedia/puppet-cargomedia/pull/1214/files#diff-a9afb52964c3a1204622d9848dd524bcR14

**Comment by ppp0** [comment]
I saw it, sorry...
Regarding your question.. How about passing a list of modules to `pulseaudio::service`, such as
```pp
pulseaudio::service { 'foo':
  user => 'bob',
  modules => [ 'bluetooth', 'x11'],
}
```

?

**Pull request merged**